frontend/test/metabase/components/PasswordReveal.unit.spec.js

```diff
@@ -1,25 +1,24 @@
-import { click } from "__support__/enzyme";
-
 import React from "react";
+import "@testing-library/jest-dom/extend-expect";
+import { render, fireEvent, cleanup } from "@testing-library/react";
 import PasswordReveal from "metabase/components/PasswordReveal";
-import CopyButton from "metabase/components/CopyButton";
 
-import { shallow } from "enzyme";
+// This shouldn't be needed from version 9.0.0 of react-testing-library (TODO: remove after update)
+afterEach(cleanup);
 
 describe("password reveal", () => {
-  let wrapper;
-
-  beforeEach(() => {
-    wrapper = shallow(<PasswordReveal />);
-  });
-
   it("should toggle the visibility state when hide / show are clicked", () => {
-    expect(wrapper.state().visible).toEqual(false);
-    click(wrapper.find("a"));
-    expect(wrapper.state().visible).toEqual(true);
+    const { getByTestId } = render(<PasswordReveal />);
+    const toggle = getByTestId("visibility-toggle");
+
+    expect(toggle.textContent).toEqual("Show");
+    fireEvent.click(toggle);
+    expect(toggle.textContent).toEqual("Hide");
```

Comment:
What do you think about removing the test id and relying on the text? 
```js
fireEvent.click(getByText("Show"));
getByText("Hide");
```

Reply:
@paulrosenzweig rebased with those changes. PR is ready now.

Reply:
Sweet. I prefer the implicit approach here and I hope it's easy to follow since we use the same style across Cypress and our (few) RTL tests.

Reply:
> rebased with those changes. PR is ready now.

I forget if external contributors can merge once a PR is approved. If so, feel free to do so. Otherwise, I can do it for you.

Reply:
I don't think so. :(
> Only those with write access to this repository can merge pull requests.

Anyway, CircleCI failed again (randomly).

Reply:
Looks like it's a Cypress flake with `type` ( https://github.com/cypress-io/cypress/issues/5480 ). I've been trying to repro this in an isolated fashion but have yet to do so. :(

Reply:
@dacort - thanks for linking to this. I'll keep an eye on it.

It isn't clear if this issue appears only in `^3.5+`. I will check for possible breaking changes in the latest version of Cypress, as it would be nice to clean up and upgrade the whole test suite.